Preview Movie Rec Tutorial

briacht opened the pull request:
## Summary

Movie Recommender Tutorial

[Docs Internal Review URL](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/movie-recommmendation?branch=pr-en-us-11329)

Commit:
Create MovieRecommender Tutorial

Commit:
Fixed media link

Commit:
Updated xrefs

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

JRAlexander committed:
fix media link

JRAlexander committed:
Add internal link

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Delete datatable.png

Commit:
Add files via upload

Commit:
Delete datatable.png

Commit:
Add files via upload

Commit:
Delete datatable.png

Commit:
Add files via upload

Commit:
Add files via upload

Commit:
swapped out image for datatable

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Delete datatable.png

Commit:
Add files via upload

Commit:
Add files via upload

Commit:
Add files via upload

Commit:
updated images

evalmetrics and dataset

Commit:
Delete datatable2.png

Commit:
Delete datatable3.png

Commit:
Delete evalmetrics.png

Commit:
Add files via upload

Commit:
Add files via upload

Commit:
Add files via upload

Commit:
Update movie-recommmendation.md

Commit:
Delete copytooutputdir.gif

Commit:
Delete copyifnewer.gif

Commit:
Delete copytooutput.gif

Commit:
Add files via upload

Commit:
Update movie-recommmendation.md

Commit:
Delete copytooutputdirectory.gif

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

JRAlexander committed:
Changed ReadFromText to LoadFromText

JRAlexander committed:
added clarification for score 3.5 check.

Commit:
add explanation of how to download data

JRAlexander committed:
Add xrefs

JRAlexander committed:
fixed formatting

JRAlexander committed:
Formatting xrefs

Commit:
update rmse and rsquared

Commit:
uncomment snippets

Commit:
remove code blocks

Commit:
update download data instructions

Commit:
move cross validation to Improve your model

Commit:
minor updates

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
update snippets

Commit:
various updates

Commit:
move explanations above code

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
renamed to recommended resources and removed other scenarios

Commit:
test

Commit:
update to test

Commit:
update test

Commit:
update link

Commit:
remove en-us, fix curly braces

Commit:
add .md

Commit:
italicize MovieRatingData.cs

Commit:
add Note for LoadModel return

Commit:
Update movie-recommmendation.md

Commit:
Update movie-recommmendation.md

Commit:
fix typo

Commit:
add image

Commit:
updates based on user study feedback

Commit:
fix image link

Commit:
fix #iterations

Commit:
a few improvements

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+
+[!code-csharp[AddAlgorithm](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#AddAlgorithm "Add the training algorithm with options")]
+
+The [MatrixFactorizationTrainer](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization%28Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options%29) as is your recommendation training algorithm.  [Matrix Factorization](https://en.wikipedia.org/wiki/Matrix_factorization_(recommender_systems)) is a common approach to recommendation when you have data on how users have  ...
```
Small typo here: need to remove the "as" from "as is your recommendation training algorithm"

natke replied:
Also, need to update [Recommended Resources](#recommended-resources) --> [Resources](#resources)

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+
+## Build and train your model
+
+There are three major concepts in ML.NET: [Data](../basic-concepts-model-training-in-mldotnet.md#data), [Transformers](../basic-concepts-model-training-in-mldotnet.md#transformer), and [Estimators](../basic-concepts-model-training-in-mldotnet.md#estimator).
```
I think we can improve this section, but we can do that post-publication.

briacht replied:
Agreed!

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options) including:
```
Can we move this section into the "Improve your model" section. We might lose some users in the explanation at this point.

briacht replied:
I think we can, but there will be the Options/hyperparameters visible in the code snippet, so users might wonder about that. I moved it down below and just added a "learn more below" link, let me know what you think

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+
+Add the following as the next line of code in the `Main()` method to call your `EvaluateModel()` method:
+
+[!code-csharp[EvaluateModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModelMain "Add EvaluateModel method in Main")]
```
Do we need to move this up above the output sections? i.e. we won't see the output until this method is called.

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+
+Add the following as the next line of code in the `Main()` method to call your `UseModelForSinglePrediction()` method:
+
+[!code-csharp[UseModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#UseModelMain "Add UseModelForSinglePrediction method in Main")]
```
Would we want to show the output of this method?

natke reviewed (changes requested):
Hi @briacht, I have made a few minor comments. Other than that, looks good!

Commit:
Update movie-recommmendation.md

Commit:
rearrange output statements

-move evaluate output after call to evaluatemodel() method in main
-add output for usemodelforsingleprediction()

Commit:
update saveModel()

Commit:
update Results / final output

Commit:
acrolinx

Commit:
Delete copytoout.gif

Commit:
add gif

natke commented on docs/machine-learning/tutorials/movie-recommmendation.md:
```diff
@@ -0,0 +1,474 @@
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options, which you can read more about in the [Algorithm hyperparameters](#algorithm-hyperparameters) section below.
```
One tiny typo '['.

Commit:
fix typo
